crates/cubecl-linalg/src/convolution/launch.rs

```diff
@@ -25,28 +25,28 @@ type Output<Alg, MP> =
     <<Alg as Algorithm>::Args as MatmulArgs>::Output<<MP as MatmulPrecision>::EO>;
 
 #[derive(Clone)]
-pub struct ConvolutionArgs {
-    pub stride: (usize, usize),
-    pub padding: (usize, usize),
-    pub dilation: (usize, usize),
+pub struct ConvolutionArgs<const N_SPATIAL: usize> {
+    pub stride: [usize; N_SPATIAL],
+    pub padding: [usize; N_SPATIAL],
+    pub dilation: [usize; N_SPATIAL],
 }
 
-/// Perform a 2D convolution using the implicit GEMM (im2col) algorithm, using cubecl tiling matmul
-/// components, using the specified algorithm.
+/// Perform an n-dimensional convolution using the implicit GEMM (im2col) algorithm, using cubecl
+/// tiling matmul components, using the specified algorithm.
 ///
-/// * `input` - The input feature map, layout should be [batches, height, width, in_channels]
-/// * `weight` - The weights (filter) applied to each kernel, layout should be [out_channels, kernel_h, kernel_w, in_channels]
-/// * `out` - The output feature map, layout should be [batches, out_height, out_width, out_channels]
+/// * `input` - The input feature map, layout should be [batches, depth, height, width, in_channels]
+/// * `weight` - The weights (filter) applied to each kernel, layout should be [out_channels, kernel_d, kernel_h, kernel_w, in_channels]
+/// * `out` - The output feature map, layout should be [batches, out_depth, out_height, out_width, out_channels]
 /// * `bias` - The bias added to each out channel
 /// * `options` - The options to use for the convolution
 #[allow(clippy::result_large_err)]
-pub fn launch_conv2d_nhwc<R: Runtime, MP: MatmulPrecision, Alg: Algorithm>(
+pub fn launch_conv<R: Runtime, MP: MatmulPrecision, Alg: Algorithm, const N_SPATIAL: usize>(
     client: &ComputeClient<R::Server, R::Channel>,
     input: &TensorHandleRef<'_, R>,
     weight: &TensorHandleRef<'_, R>,
     bias: &Option<TensorHandleRef<'_, R>>,
     out: &TensorHandleRef<'_, R>,
-    args: ConvolutionArgs,
+    args: ConvolutionArgs<N_SPATIAL>,
 ) -> Result<(), ConvLaunchError>
```

Comment:
Not sure how much the const generic affects compilation speed, but maybe creating a function without it and redispatching to it would help.